locator/token_metadata.cc

```diff
@@ -774,122 +734,114 @@ token_metadata_impl::has_pending_ranges(sstring keyspace_name, inet_address endp
-          }
-        }
-        seastar::thread::maybe_yield();
-    }
-    metadata.clear_gently().get();
-    tlogger.debug("In calculate_pending_ranges: affected_ranges.size={} ends", affected_ranges_size);
-}
-
-// Called from a seastar thread
-void token_metadata_impl::calculate_pending_ranges_for_replacing(
-        const token_metadata& unpimplified_this,
-        const abstract_replication_strategy& strategy,
-        std::unordered_multimap<range<token>, inet_address>& new_pending_ranges) const {
-    if (_replacing_endpoints.empty()) {
-        return;
-    }
-    for (const auto& node : _replacing_endpoints) {
-        auto existing_node = node.first;
-        auto replacing_node = node.second;
-        auto address_ranges = strategy.get_ranges(existing_node, unpimplified_this).get0();
-        for (const dht::token_range& x : address_ranges) {
-            seastar::thread::maybe_yield();
-            tlogger.debug("Node {} replaces {} for range {}", replacing_node, existing_node, x);
-            new_pending_ranges.emplace(x, replacing_node);
-        }
-    }
-}
-
-// Called from a seastar thread
-void token_metadata_impl::calculate_pending_ranges_for_bootstrap(
-        const abstract_replication_strategy& strategy,
-        std::unordered_multimap<range<token>, inet_address>& new_pending_ranges,
-        mutable_token_metadata_ptr all_left_metadata, dc_rack_fn& get_dc_rack) const {
-    // For each of the bootstrapping nodes, simply add and remove them one by one to
-    // allLeftMetadata and check in between what their ranges would be.
-    std::unordered_multimap<inet_address, token> bootstrap_addresses;
-    for (auto& x : _bootstrap_tokens) {
-        bootstrap_addresses.emplace(x.second, x.first);
-    }
-
-    // TODO: share code with unordered_multimap_to_unordered_map
-    std::unordered_map<inet_address, std::unordered_set<token>> tmp;
-    for (auto& x : bootstrap_addresses) {
-        auto& addr = x.first;
-        auto& t = x.second;
-        tmp[addr].insert(t);
-    }
-    for (auto& x : tmp) {
-        auto& endpoint = x.first;
-        auto& tokens = x.second;
-        all_left_metadata->update_topology(endpoint, get_dc_rack(endpoint), node::state::joining);
-        all_left_metadata->update_normal_tokens(tokens, endpoint).get();
-        auto address_ranges = strategy.get_ranges(endpoint, *all_left_metadata).get0();
-        for (const dht::token_range& x : address_ranges) {
-            new_pending_ranges.emplace(x, endpoint);
-        }
-        all_left_metadata->_impl->remove_endpoint(endpoint);
-    }
-    all_left_metadata->_impl->sort_tokens();
-}
-
 future<> token_metadata_impl::update_pending_ranges(
         const token_metadata& unpimplified_this,
-        const abstract_replication_strategy& strategy, const sstring& keyspace_name, dc_rack_fn& get_dc_rack) {
+        const abstract_replication_strategy& strategy, const sstring& keyspace_name, dc_rack_fn& get_dc_rack,
+        bool request_read_new) {
```

Comment:
nit: coding style violation.
new param should either be added to the previous line (I'd prefer that) or start with double-indent.

Reply:
All the parameters are double-indented already.